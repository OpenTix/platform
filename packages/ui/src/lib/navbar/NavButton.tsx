import styled, { css } from 'styled-components';
import React, { ButtonHTMLAttributes, FC } from 'react';

const StyledButton = styled.button<NavButtonProps>`
	margin: 0 8px 0 0;
	padding: 8px 16px;
	font-size: 16px;
	cursor: pointer;
	border-radius: 4px;
	transition: background-color 0.2s ease;
	border: 2px solid #007bff;

	${({ variant }) =>
		variant === 'filled'
			? css`
					background-color: #007bff;
					color: #ffffff;
					&:hover {
						background-color: #0056b3;
					}
			  `
			: css`
					background-color: transparent;
					color: #007bff;
					&:hover {
						background-color: rgba(0, 123, 255, 0.1);
					}
			  `}
`;

export interface NavButtonProps
	extends ButtonHTMLAttributes<HTMLButtonElement> {
	variant?: 'filled' | 'outline';
	children: React.ReactNode;
}

export const NavButton: FC<NavButtonProps> = ({
	variant = 'filled',
	children,
	...props
}) => {
	return (
		<StyledButton variant={variant} {...props}>
			{children}
		</StyledButton>
	);
};
