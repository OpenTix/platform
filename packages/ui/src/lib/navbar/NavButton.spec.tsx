import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import { NavButton } from './NavButton';

// Adjust path as needed

describe('NavButton', () => {
	it('renders with default (filled) variant', () => {
		render(<NavButton>Click Me</NavButton>); // Default variant is 'filled'
		const button = screen.getByRole('button', { name: /click me/i });
		expect(button).toBeInTheDocument();
		// Optionally check styling or class if needed
	});

	it('renders with outline variant', () => {
		render(<NavButton variant="outline">Outline Button</NavButton>);
		const button = screen.getByRole('button', { name: /outline button/i });
		expect(button).toBeInTheDocument();
		// Optionally check styling or class if needed
	});

	it('calls onClick handler', () => {
		const handleClick = jest.fn();
		render(<NavButton onClick={handleClick}>Click Handler Test</NavButton>);
		const button = screen.getByRole('button', {
			name: /click handler test/i
		});
		fireEvent.click(button);
		expect(handleClick).toHaveBeenCalledTimes(1);
	});

	it('disables button when disabled prop is passed', () => {
		render(<NavButton disabled>Disabled Button</NavButton>);
		const button = screen.getByRole('button', { name: /disabled button/i });
		expect(button).toBeDisabled();
	});

	it('matches snapshot', () => {
		const { asFragment } = render(<NavButton>Snapshot Test</NavButton>);
		expect(asFragment()).toMatchSnapshot();
	});
});
