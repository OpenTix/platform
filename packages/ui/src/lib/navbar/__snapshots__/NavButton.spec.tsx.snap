// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavButton matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="NavButton__StyledButton-sc-1y55c1v-0 dAvjaB"
  >
    Snapshot Test
  </button>
</DocumentFragment>
`;
